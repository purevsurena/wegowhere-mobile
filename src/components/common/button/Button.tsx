import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Colors } from '@/constants';
import { Click } from '@/components/layouts';
import { mScale, scale } from '@/utils/Metrics';
import { Heading } from '@/components/common/typography';
import { ButtonProps } from '@/types/button';

const Button = ({ onPress, title, disabled }: ButtonProps) => {
  return (
    <Click style={styles.button} onPress={onPress} disabled={disabled}>
      <Heading type="Heading3" label={title} color="white" />
    </Click>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    width: '100%',
    height: scale(45),
    borderRadius: mScale(22.5),
    backgroundColor: Colors.default.primary,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
