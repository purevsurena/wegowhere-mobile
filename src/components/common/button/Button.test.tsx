import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '@/components/common/button/Button';

describe('Button Component', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Button onPress={() => {}} title="Press Me" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when button is pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button onPress={onPressMock} title="Press Me" />,
    );
    fireEvent.press(getByText('Press Me'));
    expect(onPressMock).toHaveBeenCalled();
  });

  it('renders correctly when disabled', () => {
    const { toJSON } = render(
      <Button onPress={() => {}} title="Press Me" disabled />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('does not call onPress when disabled button is pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button onPress={onPressMock} title="Press Me" disabled />,
    );
    fireEvent.press(getByText('Press Me'));
    expect(onPressMock).not.toHaveBeenCalled();
  });
});
