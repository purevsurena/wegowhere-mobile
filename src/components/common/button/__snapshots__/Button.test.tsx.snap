// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4AD8DA",
      "borderRadius": 33.75,
      "height": 90,
      "justifyContent": "center",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": "100%",
    }
  }
  testID="undefined"
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "FC Subject [Non-commercial] Bd",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "700",
          "lineHeight": 22.4,
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
  >
    Press Me
  </Text>
</View>
`;

exports[`Button Component renders correctly when disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4AD8DA",
      "borderRadius": 33.75,
      "height": 90,
      "justifyContent": "center",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": "100%",
    }
  }
  testID="undefined"
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "FC Subject [Non-commercial] Bd",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "700",
          "lineHeight": 22.4,
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
  >
    Press Me
  </Text>
</View>
`;
